internal/controller/serving/inferenceservice_controller_test.go

```diff
@@ -643,115 +644,125 @@ var _ = Describe("InferenceService Controller", func() {
+			It("[serverless] if the runtime is not supported for metrics, it should create a configmap with the unsupported config", func() {
 				_ = createServingRuntime(testNs, UnsupprtedMetricsServingRuntimePath)
 				_ = createInferenceService(testNs, UnsupportedMetricsInferenceServiceName, UnsupportedMetricsInferenceServicePath)
 
-				metricsConfigMap, err := testutils.WaitForConfigMap(k8sClient, testNs, UnsupportedMetricsInferenceServiceName+constants.KserveMetricsConfigMapNameSuffix, 30, 1*time.Second)
-				Expect(err).NotTo(HaveOccurred())
-				Expect(metricsConfigMap).NotTo(BeNil())
+				verifyConfigMap(UnsupportedMetricsInferenceServiceName, testNs, false, "")
+			})
 
-				expectedMetricsConfigMap := &corev1.ConfigMap{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      UnsupportedMetricsInferenceServiceName + constants.KserveMetricsConfigMapNameSuffix,
-						Namespace: testNs,
-					},
-					Data: map[string]string{
-						"supported": "false",
-					},
-				}
-				Expect(testutils.CompareConfigMap(metricsConfigMap, expectedMetricsConfigMap)).Should(BeTrue())
+			It("[raw] if the runtime is not supported for metrics, it should create a configmap with the unsupported config", func() {
+				_ = createServingRuntime(testNs, UnsupprtedMetricsServingRuntimePath)
+				_ = createInferenceService(testNs, UnsupportedMetricsInferenceServiceName, UnsupportedMetricsInferenceServicePath, true)
+
+				verifyConfigMap(UnsupportedMetricsInferenceServiceName, testNs, false, "")
 			})
 
-			It("if the isvc does not have a runtime specified, an unsupported metrics configmap should be created", func() {
+			It("[serverless] if the isvc does not have a runtime specified, an unsupported metrics configmap should be created", func() {
 				_ = createInferenceService(testNs, NilRuntimeInferenceServiceName, NilRuntimeInferenceServicePath)
 
-				metricsConfigMap, err := testutils.WaitForConfigMap(k8sClient, testNs, NilRuntimeInferenceServiceName+constants.KserveMetricsConfigMapNameSuffix, 30, 1*time.Second)
-				Expect(err).NotTo(HaveOccurred())
-				Expect(metricsConfigMap).NotTo(BeNil())
+				verifyConfigMap(NilRuntimeInferenceServiceName, testNs, false, "")
+			})
 
-				expectedmetricsConfigMap := &corev1.ConfigMap{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      NilRuntimeInferenceServiceName + constants.KserveMetricsConfigMapNameSuffix,
-						Namespace: testNs,
-					},
-					Data: map[string]string{
-						"supported": "false",
-					},
-				}
-				Expect(testutils.CompareConfigMap(metricsConfigMap, expectedmetricsConfigMap)).Should(BeTrue())
+			It("[raw] if the isvc does not have a runtime specified, an unsupported metrics configmap should be created", func() {
+				_ = createInferenceService(testNs, NilRuntimeInferenceServiceName, NilRuntimeInferenceServicePath, true)
+
+				verifyConfigMap(NilRuntimeInferenceServiceName, testNs, false, "")
```

Comment:
if isvc does not set runtime, there are 2 available returns
1. `return desiredServingRuntime, nil`
2. `return desiredServingRuntime, errors.New(constants.NoSuitableRuntimeError)`

For the first return, it will create a `supported` configmap. So the title would be changed to like
"[raw] if the isvc does not have a runtime specified and there is no supported runtimes, an unsupported metrics configmap should be created"

By the way, we don't support dynamic runtime selection so I am not sure if we should provide this kind of finding supported runtime feature for metrics. runtime field in isvc is a kind of mandatory.